lfr/heterogeneous_single_assembly_3D/neutronics_standalone/neutronics.i

```diff
@@ -0,0 +1,1771 @@
+    duct_intervals = '1 1'
+    background_block_id = '4'
+    duct_block_ids = '5 6'
+    external_boundary_id = 997
+    pattern = '1 1 1 1 1 1 1;
+              1 0 0 0 0 0 0 1;
+             1 0 0 0 0 0 0 0 1;
+            1 0 0 0 0 0 0 0 0 1;
+           1 0 0 0 0 0 0 0 0 0 1;
+          1 0 0 0 0 0 0 0 0 0 0 1;
+         1 0 0 0 0 0 0 0 0 0 0 0 1;
+          1 0 0 0 0 0 0 0 0 0 0 1;
+           1 0 0 0 0 0 0 0 0 0 1;
+            1 0 0 0 0 0 0 0 0 1;
+             1 0 0 0 0 0 0 0 1;
+              1 0 0 0 0 0 0 1;
+               1 1 1 1 1 1 1'
+  []
+  [coarse_mesh]
+    type = AdvancedExtruderGenerator
+    input = ASM_CM
+    heights = '${dhA} ${dhB} ${dhC} ${dhD} ${dhE} ${dhF} ${dhG} ${dhH} ${dhI} ${dhJ}'
+    num_layers = '${num_axmeshA} ${num_axmeshB} ${num_axmeshC} ${num_axmeshD} ${num_axmeshE} ${num_axmeshF} ${num_axmeshG} ${num_axmeshH} ${num_axmeshI} ${num_axmeshJ}'
+    direction = '0 0 1'
+    top_boundary = 998
+    bottom_boundary = 999
+  []
+
+  [cmesh]
+    type = CoarseMeshExtraElementIDGenerator
+    input = assign
+    coarse_mesh = coarse_mesh
+    extra_element_id_name = coarse_element_id
+  []
+  uniform_refine = 0
+[]
+
+# ==============================================================================
+# Transport Systems
+# ==============================================================================
+[TransportSystems]
+  particle = neutron
+  equation_type = eigenvalue
+  G = 9
+  VacuumBoundary = 'ASSEMBLY_TOP ASSEMBLY_BOTTOM'
+  ReflectingBoundary = 'ASSEMBLY_SIDE'
+  [sn]
+    scheme = DFEM-SN
+    family = MONOMIAL
+    order = FIRST
+    AQtype = Gauss-Chebyshev
+    NPolar = 2
+    NAzmthl = 3
+    NA = 1
+    sweep_type = asynchronous_parallel_sweeper
+    using_array_variable = true
+    collapse_scattering  = true
+  []
+[]
+
+# ==============================================================================
+# Executioner
+# ==============================================================================
+[Executioner]
+  type = SweepUpdate
+  verbose = true
+
+  richardson_rel_tol = 1e-4
+  richardson_max_its = 500
+  richardson_value = 'eigenvalue'
+  inner_solve_type = SI
+  max_inner_its = 9
+
+  cmfd_acceleration = true
+  coarse_element_id = coarse_element_id
+  cmfd_eigen_solver_type = krylovshur
+[]
+
+# ==============================================================================
+# Flux Normalization for Total Power
+# ==============================================================================
+[PowerDensity]
+  power = ${totalpower}
+  power_density_variable = power_density
+  family = MONOMIAL
+  order = CONSTANT
+[]
+
+# ==============================================================================
+# Fuel pin-wise and axial mesh-wise power output
+# ==============================================================================
+[AuxVariables]
+  [volume]
+    family = MONOMIAL
+    order = CONSTANT
+    initial_condition = 1
+  [] 
+  [x_coord]
+    family = LAGRANGE
+    order = FIRST
```

Comment:
You can use subblock here
```
[InitialCondition]
  type = FunctionIC
  function = x
[]
```
to add IC and do not have to have a separate IC block.

Reply:
So, I see the error message `section '[AuxVariables/x_coord/ICx]' does not have an associated "Action".`
Isn't the below what you've recommended?
```
  [x_coord]
    family = LAGRANGE
    order = FIRST
    [ICx]
      type = FunctionIC
      function = x
    []
  []
```

Reply:
Oh.... `InitialCondition` is not a user-defined name..

Reply:
Right. It has to be that.